test/v2/services/XmlParsersSpec.scala

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright 2022 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package v2.services
+
+import akka.stream.scaladsl.Sink
+import org.scalatest.concurrent.ScalaFutures
+import org.scalatest.freespec.AnyFreeSpec
+import org.scalatest.matchers.must.Matchers
+import org.scalatestplus.scalacheck.ScalaCheckPropertyChecks
+import v2.base.StreamTestHelpers
+import v2.base.TestActorSystem
+import v2.models.errors.ExtractionError
+import v2.models.request.MessageType
+
+import scala.xml.NodeSeq
+
+class XmlParsersSpec extends AnyFreeSpec with TestActorSystem with Matchers with StreamTestHelpers with ScalaFutures with ScalaCheckPropertyChecks {
+  "MessageType parser" - {
```

Comment:
If you're calling it a MessageTypeExtractor, use that term please so people can search for that term through the tests

Reply:
done